vcr/verifier/signature_verifier.go

```diff
@@ -116,7 +120,19 @@ func (sv *signatureVerifier) jwtSignature(jwtDocumentToVerify string, issuer str
 	var keyID string
 	_, err := crypto.ParseJWT(jwtDocumentToVerify, func(kid string) (crypt.PublicKey, error) {
 		keyID = kid
-		return sv.resolveSigningKey(kid, issuer, at)
+		metadata := &resolver.ResolveMetadata{
+			ResolveTime: at,
+		}
+		if strings.HasPrefix(keyID, "did:x509:") {
+			message, _ := jws.ParseString(jwtDocumentToVerify)
+			if message != nil && len(message.Signatures()) > 0 {
+				headers := message.Signatures()[0].ProtectedHeaders()
+				metadata.X509CertThumbprint = headers.X509CertThumbprint()
+				metadata.X509CertChain = headers.X509CertChain()
+				metadata.X509CertThumbprintS256 = headers.X509CertThumbprintS256()
+			}
+		}
```

Comment:
why not always do this, regardless of the DID method? Probably less complex.

Although: you could decide to only do this when it's did:x509, then you can (more) safely return an error when `ParseString()` fails. Since the x5c chain is required for the process, I think it should fail if it can't parse the JWS? (or maybe even when x5c isn't there?)

Reply:
It does not make a lot of sense to do this whenever the method is not x509, but somehow understandable if you would fill the headers anyway. There MIGHT be a situation where the fields are used outside the x509 spec..

About failing: 
I think it will fail later on when X509CertThumbprint et. al. is not filled and the method is x509.

Reply:
Maybe the cleanest way would be to just stuff in the headers into the metadata. That would be way more generic.